x-pack/heartbeat/monitors/browser/source/pushed.go

```diff
@@ -0,0 +1,131 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package source
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/elastic/elastic-agent-libs/mapstr"
+)
+
+type PushedSource struct {
+	Content         string `config:"content" json:"content"`
+	TargetDirectory string
+}
+
+var ErrNoContent = fmt.Errorf("no 'content' value specified for pushed monitor source")
+
+func (p *PushedSource) Validate() error {
+	if !regexp.MustCompile(`\S`).MatchString(p.Content) {
+		return ErrNoContent
+	}
+
+	return nil
+}
+
+func (p *PushedSource) Fetch() error {
+	decodedBytes, err := base64.StdEncoding.DecodeString(p.Content)
+	if err != nil {
+		return err
+	}
+
+	tf, err := ioutil.TempFile("/tmp", "elastic-synthetics-zip-")
```

Comment:
Shouldn't we also `defer` the `close` of this file? Although we're already removing it [the docs don't mention `Remove` closes the open handle](https://pkg.go.dev/os#Remove).

PS: [I found it interesting to learn how these are made to be globally unique](https://stackoverflow.com/a/61050527)

Reply:
Its explicity handled after the monitor run to completion, we control the close using `p.Close()` on any error.